Tests.Unit.Presentation.Web/Models/V2/ItContractWriteModelMapperTest.cs

```diff
@@ -554,6 +559,109 @@ private static void AssertRoles(List<RoleAssignmentRequestDTO> roles, List<UserR
             }
 		}
 
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Can_Map_PaymentModel(bool hasValues)
+        {
+            //Arrange
+            var milestones = hasValues ? Many<PaymentMileStoneDTO>().ToList() : null;
+            var input = new ContractPaymentModelDataWriteRequestDTO()
+            {
+                OperationsRemunerationStartedAt = hasValues ? A<DateTime>() : null,
+                PaymentFrequencyUuid = hasValues ? A<Guid>() : null,
+                PaymentModelUuid = hasValues ? A<Guid>() : null,
+                PriceRegulationUuid = hasValues ? A<Guid>() : null,
+                PaymentMileStones = milestones
+            };
+
+            //Act
+            var output = _sut.MapPaymentModel(input);
+
+            //Assert
+            AssertPaymentModel(input, output, hasValues);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Can_Map_PaymentModel_FromPOST(bool hasValues)
+        {
+            //Arrange
+            var milestones = hasValues ? Many<PaymentMileStoneDTO>().ToList() : null;
+            var input = new ContractPaymentModelDataWriteRequestDTO()
+            {
+                OperationsRemunerationStartedAt = hasValues ? A<DateTime>() : null,
+                PaymentFrequencyUuid = hasValues ? A<Guid>() : null,
+                PaymentModelUuid = hasValues ? A<Guid>() : null,
+                PriceRegulationUuid = hasValues ? A<Guid>() : null,
+                PaymentMileStones = milestones
+            };
+
+            //Act
+            var output = _sut.FromPOST(new CreateNewContractRequestDTO() { PaymentModel = input });
+
+            //Assert
+            AssertPaymentModel(input, output.PaymentModel.Value, hasValues);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Can_Map_PaymentModel_FromPUT(bool hasValues)
+        {
+            //Arrange
+            var milestones = hasValues ? Many<PaymentMileStoneDTO>().ToList() : null;
+            var input = new ContractPaymentModelDataWriteRequestDTO()
+            {
+                OperationsRemunerationStartedAt = hasValues ? A<DateTime>() : null,
+                PaymentFrequencyUuid = hasValues ? A<Guid>() : null,
+                PaymentModelUuid = hasValues ? A<Guid>() : null,
+                PriceRegulationUuid = hasValues ? A<Guid>() : null,
+                PaymentMileStones = milestones
+            };
+
+            //Act
+            var output = _sut.FromPUT(new UpdateContractRequestDTO { PaymentModel = input });
+
+            //Assert
+            AssertPaymentModel(input, output.PaymentModel.Value, hasValues);
+        }
+
+        private static void AssertPaymentModel(ContractPaymentModelDataWriteRequestDTO input, ItContractPaymentModelModificationParameters output, bool hasValues)
+        {
+            if (hasValues)
```

Comment:
hvis du bruger null propagation behøves du ikke gentage begge blokke. Dine assertions vil jo stadig være korrekte

Reply:
DateTime og Listen omskrives til Maybe. Så hvis der ikke er en værdi så er de Maybe.None. Så de to er jeg nød til at teste på to forskellige måder

Reply:
ikke hvis du bruger match

Reply:
altså i din expectation

Reply:
så vil du kunne nøjes med 1 blok